src/libcrun/utils.c

```diff
@@ -712,6 +712,16 @@ check_running_in_user_namespace (libcrun_error_t *err)
   if (ret >= 0)
     return ret;
 
+  // handle kernels with user namespaces disabled
+  ret = crun_path_exists ("/proc/self/uid_map", err);
+  if (UNLIKELY (ret < 0))
+      return ret;
+  else if (UNLIKELY (ret == 0))
+    {
+      run_in_userns = 0;
+      return run_in_userns;
+    }
+
   ret = read_all_file ("/proc/self/uid_map", &buffer, &len, err);
   if (UNLIKELY (ret < 0))
     return ret;
```

Comment:
we need to treat ENOENT as `run_in_userns = 0;` so we don't need an additional syscall

Reply:
sorry, I am not sure if I fully understand your comment

The check needs to happen before `read_all_file` call, otherwise `crun_make_error` will be called on non-existing file.